source/simulator/solver.cc

```diff
@@ -203,6 +203,8 @@ namespace aspect
 
         unsigned int n_iterations_A() const;
         unsigned int n_iterations_S() const;
+        void set_A_block_tolerance(double temp_A_block_tolerance) const;
+        void set_S_block_tolerance(double temp_S_block_tolerance) const;
```

Comment:
Is there a reason you do not put the tolerances as additional arguments into the constructor? Then we could save the additional functions and also make the tolerances `const double` instead of `mutable double` like the `do_solve_a` variable. `n_Iterations_A_` and `S` are changed by the solver itself from the outside and I guess therefore they need to be mutable, but the tolerances does not really fulfil the purpose of a mutable of being something that is not so important to the object itself, but for others.